### PR TITLE
[Target] Don't crash trying to dereferencing a nullptr.

### PR DESCRIPTION
**Pull request by dcci**
Fixes <rdar://problem/55701649>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

**Comment by dcci** [comment]
@swift-ci test

File: lldb/source/Target/Target.cpp [hunk 1]
```diff
@@ -2251,12 +2251,16 @@ Target::GetScratchTypeSystemForLanguage(lldb::LanguageType language,
       }
     } else if (create_on_demand) {
       if (StreamSP error_stream_sp = GetDebugger().GetAsyncErrorStream()) {
+        auto module = GetExecutableModule();
+        const char *module_name = nullptr;
+        if (module_name)
```
Comment (adrian-prantl) [inline lldb/source/Target/Target.cpp]: That looks like a trivially false condition?

**Review by adrian-prantl — CHANGES_REQUESTED** [review]
I think this doesn't work :-)

File: lldb/source/Target/Target.cpp (hunk 1 above)
Comment (dcci) [reply lldb/source/Target/Target.cpp]: Sigh, that should be module :(

Comment (dcci) [reply lldb/source/Target/Target.cpp]: as in `if (module)`

**Comment by dcci** [comment]
Now with the correct condition.

**Comment by dcci** [comment]
@swift-ci test

File: lldb/source/Target/Target.cpp [hunk 2]
```diff
@@ -2251,12 +2251,16 @@ Target::GetScratchTypeSystemForLanguage(lldb::LanguageType language,
...
+                            ->GetPlatformFileSpec()
+                            .GetFilename()
+                            .AsCString();
         error_stream_sp->Printf(
             "Shared Swift state for %s could not be initialized.\n",
-            GetExecutableModule()
-                ->GetPlatformFileSpec()
-                .GetFilename()
-                .AsCString());
+            module_name);
```
Comment (adrian-prantl) [inline lldb/source/Target/Target.cpp:2263]: `%s` with a `nullptr`  does not crash?

Comment (adrian-prantl) [reply lldb/source/Target/Target.cpp:2263]: What about
`const char *module_name = "[no executable module]";`

**Comment by dcci** [comment]
@adrian-prantl addressed your feedback.

**Comment by dcci** [comment]
@swift-ci test

**Review by adrian-prantl — APPROVED** [review]

**Pull request merged**